File: backend/app/schemas/review.py
"""Pydantic schemas for /api/v1/reviews/* endpoints (Phase 3)."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class MentionOut(BaseModel):
    """A single collaborator mention within a review."""

    model_config = ConfigDict(from_attributes=True)

    collaborator_id: int
    collaborator_name: str
    mention_snippet: str | None = None
    match_score: float | None = None


class ReviewOut(BaseModel):
    """Compact review representation for list endpoints."""

    model_config = ConfigDict(from_attributes=True)

    review_id: str
    location_id: str | None = None
    rating: int | None = None
    comment: str | None = None
    reviewer_name: str | None = None
    is_anonymous: bool | None = None
    create_time: datetime | None = None
    update_time: datetime | None = None
    reply_text: str | None = None
    reply_time: datetime | None = None
    review_url: str | None = None
    is_local_guide: bool | None = None

    # Denormalized from review_labels
    sentiment: str | None = None
    is_enotariado: bool | None = None

    # Denormalized from review_collaborators
    collaborator_names: list[str] = []


class ReviewDetailOut(ReviewOut):
    """Full review representation for single-review GET."""

    original_language: str | None = None
    translated_text: str | None = None
    response_text: str | None = None
    response_time: datetime | None = None
    reviewer_id: str | None = None
    reviewer_url: str | None = None
    reviewer_photo_url: str | None = None
    collection_source: str | None = None
    processed_at: datetime | None = None

    mentions: list[MentionOut] = []


class ReviewListResponse(BaseModel):
    """Paginated review list with cursor-based pagination."""

    items: list[ReviewOut]
    next_cursor: str | None = None
    has_more: bool
    total: int

File: frontend/src/types/review.ts
export interface ReviewOut {
  review_id: string
  location_id: string | null
  rating: number | null
  comment: string | null
  reviewer_name: string | null
  is_anonymous: boolean | null
  create_time: string | null
  update_time: string | null
  reply_text: string | null
  reply_time: string | null
  review_url: string | null
  is_local_guide: boolean | null
  sentiment: string | null
  is_enotariado: boolean | null
  collaborator_names: string[]
}

export interface MentionOut {
  collaborator_id: number
  collaborator_name: string
  mention_snippet: string | null
  match_score: number | null
}

export interface ReviewDetailOut extends ReviewOut {
  original_language: string | null
  translated_text: string | null
  response_text: string | null
  response_time: string | null
  reviewer_id: string | null
  reviewer_url: string | null
  reviewer_photo_url: string | null
  collection_source: string | null
  processed_at: string | null
  mentions: MentionOut[]
}

export interface ReviewListResponse {
  items: ReviewOut[]
  next_cursor: string | null
  has_more: boolean
  total: number
}
